bokehjs/src/lib/models/glyphs/hspan.ts

```diff
@@ -0,0 +1,156 @@
+import {SpatialIndex} from "core/util/spatial"
+import {map} from "core/util/arrayable"
+import {range} from "core/util/array"
+import * as p from "core/properties"
+
+const {abs, max} = Math
+
+export type HSpanData = GlyphData & p.UniformsOf<HSpan.Mixins> & {
+  _y: FloatArray
+  sy: ScreenArray
+
+  max_line_width: number
+}
+
+export interface HSpanView extends HSpanData {}
+
+export class HSpanView extends GlyphView {
+  declare model: HSpan
+  declare visuals: HSpan.Visuals
+
+  override after_visuals(): void {
+    super.after_visuals()
+    this.max_line_width = uniforms.max(this.line_width)
+  }
+
+  protected override _index_data(index: SpatialIndex): void {
+    for (const y_i of this._y) {
+      index.add_point(0, y_i)
+    }
+  }
+
+  protected override _bounds(bounds: Rect): Rect {
+    const {y0, y1} = bounds
+    return {x0: NaN, x1: NaN, y0, y1}
+  }
+
+  protected override _map_data(): void {
+    super._map_data()
+    const {round} = Math
+    this.sy = map(this.sy, (yi) => round(yi))
+  }
+
+  scenterxy(i: number): [number, number] {
+    const {hcenter} = this.renderer.plot_view.frame.bbox
+    return [hcenter, this.sy[i]]
+  }
+
+  protected _render(ctx: Context2d, indices: number[], data?: HSpanData): void {
+    const {sy} = data ?? this
+    const {left, right} = this.renderer.plot_view.frame.bbox
+
+    for (const i of indices) {
+      const sy_i = sy[i]
+
+      if (!isFinite(sy_i))
+        continue
+
+      ctx.beginPath()
+      ctx.moveTo(left, sy_i)
+      ctx.lineTo(right, sy_i)
+
+      this.visuals.line.apply(ctx, i)
+    }
+  }
+
+  protected _get_candidates(sy0: number, sy1?: number): Iterable<number> {
+    const {max_line_width} = this
+    const [y0, y1] = this.renderer.yscale.r_invert(sy0 - max_line_width, (sy1 ?? sy0) + max_line_width)
+    return this.index.indices({x0: 0, x1: 0, y0, y1})
+  }
+
+  protected _find_spans(candidates: Iterable<number>, fn: (sy: number, line_width: number) => boolean): number[] {
+    const {sy, line_width} = this
+    const indices: number[] = []
+
+    for (const i of candidates) {
+      const sy_i = sy[i]
+      const line_width_i = line_width.get(i)
+      if (fn(sy_i, line_width_i)) {
+        indices.push(i)
+      }
+    }
+
+    return indices
+  }
+
+  protected override _hit_point(geometry: PointGeometry): Selection {
+    const {sy: gsy} = geometry
+    const candidates = this._get_candidates(gsy)
+    const indices = this._find_spans(candidates, (sy, line_width) => abs(sy - gsy) <= max(line_width/2, 2/*px*/))
+    return new Selection({indices})
+  }
+
+  protected override _hit_span(geometry: SpanGeometry): Selection {
+    const indices = (() => {
+      if (geometry.direction == "v") {
+        return range(0, this.data_size)
+      } else {
+        const {sy: gsy} = geometry
+        const candidates = this._get_candidates(gsy)
+        return this._find_spans(candidates, (sy, line_width) => abs(sy - gsy) <= max(line_width/2, 2/*px*/))
```

Comment:
FWW I would find 
```typescript
const comparator = (sy, line_width) => abs(sy - gsy) <= max(line_width/2, 2/*px*/)
return this._find_spans(candidates, comparator)
```
much easier to read and parse for all of thse `_find_spans` calls

Reply:
This doesn't allow to infer the type of `comparator` (or its arguments) from the context. My preferred way to split such lines if necessary, is like this:
```ts
return this._find_spans(candidates, (sy, line_width) => {
  return abs(sy - gsy) <= max(line_width/2, 2/*px*/)
})
```